cengage/react-magma
# feat(Combobox, Select, Date picker, Dropdown): Update viewport detection/positioning

In the scope of this ticket, the ability to detect the viewport and position certain components was integrated via [https://floating-ui.com/](https://floating-ui.com/) for some components:

- Combobox
- Date picker
- Dropdown
- Select

Feat: [#62](https://github.com/cengage/react-magma/issues/62)
Issue: [#958](https://github.com/cengage/react-magma/issues/958)

## What I did


## Screenshots:
![Screenshot from 2024-09-24 16-10-47](https://github.com/user-attachments/assets/3f224479-e612-4d6d-aa36-82ade953dace)
![Screenshot from 2024-09-24 16-10-35](https://github.com/user-attachments/assets/71d92764-5116-484e-be86-e5e10c84af96)
![Screenshot from 2024-09-24 16-10-26](https://github.com/user-attachments/assets/f1be4521-5d27-4587-8c46-00eb2e21c2c2)
![Screenshot from 2024-09-24 16-10-03](https://github.com/user-attachments/assets/44c40a76-4869-4c1f-9a99-7053ae612fa3)
![Screenshot from 2024-09-24 16-09-48](https://github.com/user-attachments/assets/c6a579b9-228b-42e2-9733-7b33647da825)
![Screenshot from 2024-09-20 15-04-52](https://github.com/user-attachments/assets/0c5828a1-7568-4f63-bb1b-9523eeae76f3)
![Screenshot from 2024-09-20 15-03-13](https://github.com/user-attachments/assets/4e7b888a-3cc1-440b-92ad-5793876af9b6)
![Screenshot from 2024-09-20 14-03-10](https://github.com/user-attachments/assets/4d3357ba-64f2-42bf-b052-941580b97e10)
![Screenshot from 2024-09-20 12-46-12](https://github.com/user-attachments/assets/468f20cb-2dbc-4383-9d70-2fa8fdb0fa93)
![Screenshot from 2024-09-20 12-46-07](https://github.com/user-attachments/assets/72909d79-9533-4ff5-b858-40ab3e2c2baf)
![Screenshot from 2024-09-20 12-45-57](https://github.com/user-attachments/assets/9bd649e9-ec49-40b1-8f0b-191b26ae3b3b)
![Screenshot from 2024-09-19 12-34-28](https://github.com/user-attachments/assets/80678b8a-3f77-45f5-8221-748a5d5b2bbf)
![Screenshot from 2024-09-25 14-01-06](https://github.com/user-attachments/assets/119a0f40-88c9-4285-8b19-b89e909757b9)
![Screenshot from 2024-09-25 14-01-27](https://github.com/user-attachments/assets/a2091866-a858-43f9-8e35-81ccc6e8e230)
![Screenshot from 2024-09-25 14-01-21](https://github.com/user-attachments/assets/fa1dfdac-4083-4f90-a97c-135e30f2d782)
![Screenshot from 2024-09-25 14-01-13](https://github.com/user-attachments/assets/fe938137-4605-428f-88c1-d4f3030652b2)
[Screencast from 25.09.24 14:15:32.webm](https://github.com/user-attachments/assets/e8c6c68a-374b-40dc-8c2f-daae56dc9250)



## Checklist 

## How to test
Storybook -> Combobox
Storybook -> Date picker
Storybook -> Dropdown -> Flipped Items
Storybook -> Select

For all of these components, nothing should be changed.

If we want to test the new feature, we can use react magma docs and open this components in CodeSandBox to test it with some components in the case, when there isn't enough room for the Combobox, Date picker, Dropdown, and Select to display in the position it's supposed to, then it will automatically move to the opposite position.

 For example: 
  - the down position will move to the up position when there isn't enough space below the button and vice versa.
  - the right position will move to the left position when there isn't enough space and vice versa.
